src/agent/srcview/src/bin/srcview.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[macro_use]
+extern crate clap;
+
+use anyhow::{Context, Result};
+use env_logger;
+use srcview::{ModOff, Report, SrcLine, SrcView};
+use std::fs::{self};
+use std::io::{stdout, Write};
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+enum Opt {
+    DumpSrcloc(SrcLocOpt),
+    DumpPdbPaths(PdbPathsOpt),
+    DumpModoffs(ModOffOpt),
+    DumpCobertuna(DumpCobertunaOpt),
+    Licenses,
+    Version,
+}
+
+#[derive(StructOpt, Debug)]
+struct ModOffOpt {
+    modoff_path: PathBuf,
+}
+
+#[derive(StructOpt, Debug)]
+struct PdbPathsOpt {
+    pdb_path: PathBuf,
+}
+
+#[derive(StructOpt, Debug)]
+struct SrcLocOpt {
+    pdb_path: PathBuf,
+    modoff_path: PathBuf,
+}
+
+#[derive(StructOpt, Debug)]
+struct DumpCobertunaOpt {
+    pdb_path: PathBuf,
+    modoff_path: PathBuf,
+}
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    let opt = Opt::from_args();
+
+    match opt {
+        Opt::DumpSrcloc(opts) => dump_srcloc(opts)?,
+        Opt::DumpPdbPaths(opts) => dump_pdb_paths(opts)?,
+        Opt::DumpModoffs(opts) => dump_mod_offs(opts)?,
+        Opt::DumpCobertuna(opts) => dump_cobertura(opts)?,
+        Opt::Licenses => licenses()?,
+        Opt::Version => version(),
+    };
+
+    Ok(())
+}
+
+fn version() {
+    println!("{}", crate_version!());
+}
+
+fn licenses() -> Result<()> {
+    stdout().write_all(include_bytes!("../../../data/licenses.json"))?;
+    Ok(())
+}
+
+fn dump_srcloc(opts: SrcLocOpt) -> Result<()> {
+    let modoff_data = fs::read_to_string(&opts.modoff_path)
+        .with_context(|| format!("unable to read modoff_path: {}", opts.modoff_path.display()))?;
+    let modoffs = ModOff::parse(&modoff_data)?;
+    let mut srcview = SrcView::new();
+    srcview.insert("example.exe", &opts.pdb_path)?;
```

Comment:
we probably want to make this an optional argument

Reply:
The way I handle this from the code that was dump_cobertura is probably a bit better -- creating four variants for bare, dll, exe, and sys. Im unsure if we should be strict (modoff must be "foo.exe+1234") or relaxed (we accept foo.exe+1234 and foo+1234). The latter is more common from the windbg case, but there are disambiguation problems if a process has both foo.exe and foo.dll in its address space.

Reply:
Thinking about this slightly more, youre right, it should just be an argument. There's no reason we'd want to enforce that pdb name match the module it describes, despite that being convention. This also solves us from having to create multiple pdbcache's per base name without refcounting or similar.

Reply:
I went with the model where it's an optional argument.  If a value is not provided, it uses your original method of file stem + known extensions.